### PR TITLE
Reduce index zip size

### PR DESCRIPTION
**Pull request by daryakoval**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://jira-hq.paloaltonetworks.local/browse/CIAC-3872

## Description
Reduces the number of versions in the index.zip file. 
We are deleting old versions from the changelog file, and then deleting all metadata files related to the old versions.
How do we determine which version is old enough and if we need to delete it? 
We are keeping the maximum number of versions from the following: 
    1.  Versions were released last year.
    2.  Last minor version and one version before it.
    3.  Last five versions.

Testing with Packs/Base/ReleaseNotes/1_31_52.md, ignore
## Screenshots
Paste here any images that will help the reviewer

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Tests/Marketplace/marketplace_services.py [hunk 1]
```diff
@@ -4324,3 +4328,53 @@ def is_content_item_in_id_set(display_name: str, rn_header: str, id_set: dict, m
...
+    last_same_minor_versions = []
+    last_year_versions = []
+    last_five_versions = list(changelog.keys())[-5:]
+
+    year_ago_datetime_obj = datetime.utcnow() - timedelta(days=365)
+    last_version = Version(list(changelog.keys())[-1])
+    save_last_minor_versions = not (last_version.minor == 0 and last_version.major == 1)
+
+    prev_version = None
+    for version, info in changelog.items():
```
Comment (dantavori) [inline Tests/Marketplace/marketplace_services.py:4360]: let's make sure the iteration is sorted by `key=lambda v: Version(v)`

**Review by dantavori — COMMENTED** [review]
looks great 👍🏽 
@yaakovpraisler please approve

File: Tests/Marketplace/marketplace_services.py (hunk 1 above)
Comment (daryakoval) [reply Tests/Marketplace/marketplace_services.py:4360]: The iteration is sorted by the order it is in the changelog. When reading from json, the order is saved.

File: Tests/Marketplace/marketplace_services.py [hunk 2]
```diff
@@ -4324,3 +4328,53 @@ def is_content_item_in_id_set(display_name: str, rn_header: str, id_set: dict, m
...
+    """
+    versions_to_keep: List[str] = []
+    if not changelog:
+        return versions_to_keep
+
+    last_same_minor_versions = []
+    last_year_versions = []
+    last_five_versions = list(changelog.keys())[-5:]
+
+    year_ago_datetime_obj = datetime.utcnow() - timedelta(days=365)
```
[the remark this answers is not in the document]
Comment (daryakoval) [reply Tests/Marketplace/marketplace_services.py:4355]: https://docs.python.org/3/library/datetime.html#datetime.timedelta

No such option: `years`.

**Comment by daryakoval** [comment]
Link to the test upoads: 
https://code.pan.run/xsoar/content/-/pipelines/4329738

https://code.pan.run/xsoar/content/-/pipelines/4356746

**Review by dantavori — APPROVED** [review]

File: Tests/Marketplace/marketplace_constants.py [hunk 3]
```diff
@@ -186,6 +186,7 @@ class Metadata(object):
     URL = 'url'
     MARKETPLACES = 'marketplaces'
     DISABLE_MONTHLY = 'disableMonthly'
+    METADATA_FILE_REGEX_GET_VERSION = r'metadata\-([\d\.]+)\.json'
```
Comment (yaakovpraisler) [inline Tests/Marketplace/marketplace_constants.py:189]: The Metadata class is more for metadata properties, please define this variable as a separate constant var from the metadata

**Review by yaakovpraisler — COMMENTED** [review]
Looks good! Just a small comment.

**Pull request merged**